pkg/controller/gitopsconfig/controller.go

```diff
@@ -207,31 +208,51 @@ func ContainsTrigger(instance *gitopsv1alpha1.GitOpsConfig, triggeType string) b
 }
 
 // createJob creates a new gitops job for the passed instance
-func (r *Reconciler) createJob(jobtype string, instance *gitopsv1alpha1.GitOpsConfig) error {
+func (r *Reconciler) createJob(jobtype string, instance *gitopsv1alpha1.GitOpsConfig) (reconcile.Result, error) {
 	//TODO add logic to ignore if another job was created sooner than x (5 minutes?) time and it is still running.
 	mergedata := util.JobMergeData{
 		Config: *instance,
 		Action: jobtype,
 	}
+	// looking up for running jobs
+	jobList := &batchv1.JobList{}
+	err = r.client.List(context.TODO(), &client.ListOptions{
+		Namespace: instance.Namespace,
+	}, jobList)
+
+	if err != nil {
+		log.Error(err, "unable to list the jobs")
+		return reconcile.Result{}, err
+	}
+
+	for _, jobFromList := range jobList.Items {
+		if isOwner(instance, &jobFromList.ObjectMeta) {
+			if jobFromList.Status.Active != 0 {
+				log.Info("Job is runnning for this instance...." + instance.Name)
+				return reconcile.Result{Requeue: true}, nil
+			}
+		}
+
+	}
```

Comment:
Please now try to write an e2e test for this feature.

My initial idea how I would try to write the test, is:
 1. trigger eunomia multiple times quickly, for example, by [sending values to gitopsconfig.PushEvents channel in a similar way as is done in eunomia Webhook handler code](https://github.com/KohlsTechnology/eunomia/blob/e1900b95cf469cfeb45f748da9bf773ebb54b641/pkg/handler/handler.go#L100-L103)
      - if this doesn't work initially, it may be because of rate limiting in operator-sdk; if yes, it may require finding rate limiting options in operator-sdk and tweaking them
 2. verify somehow in logs that message from line 231 above was emitted, meaning previous job was detected and new one postponed
 3. verify that after first job finished, second one (that was postponed) becomes started

Reply:
unable to fetch/read the operator logs to run the e2e test.

Reply:
I discussed with with @akavel in chat ... we are not sure the best way to write unit or e2e tests for this code change, so that this time we are merging without tests being added.